lib/spack/spack/test/util/executable.py

```diff
@@ -62,24 +75,49 @@ def test_which_with_slash_ignores_path(tmpdir, working_env):
     wrong_path = str(tmpdir.join('bin', 'exe'))
     os.environ['PATH'] = os.path.dirname(wrong_path)
 
-    fs.set_executable(path)
-    fs.set_executable(wrong_path)
+    if sys.platform == "win32":
+        with tmpdir.as_cwd():
+            fs.touch("exe.exe")
+            fs.touch('bin{0}exe'.format(os.path.sep))
+            path = path + ".exe"
+            wrong_path = wrong_path + ".exe"
+            exe = ex.which('.{0}exe'.format(os.path.sep))
+            assert exe.path == path
 
-    with tmpdir.as_cwd():
-        exe = ex.which('./exe')
-        assert exe.path == path
+    else:
+        fs.set_executable(path)
+        fs.set_executable(wrong_path)
+
+        with tmpdir.as_cwd():
```

Comment:
This code block is the same for all platforms, it can be moved out of the if statement to avoid duplicate code.